torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -1173,17 +1232,21 @@ struct InterpreterStateImpl : c10::intrusive_ptr_target {
             push(stack, IValue());
             push(stack, IValue());
             push(stack, IValue());
-            runGraphFunction(stack, &f, &af);
+            runGraphFunction(stack, &f, &af, call_info);
           } break;
-          case OP:
+          case OP: {
+            TLSCallInfoGuard g(call_info);
```

Comment:
I'm a bit worried about introducing another TLS write on every op. Given that we have Frame hierarchy in TS too (just like we do in python) - can't we instead just have a TLS of "active interpreter" (which gets updated only once in runImpl) and then have a method to obtain from the active frame and it's hierarchy)? I think we'd need to move `af.pc` into `frame.pc` every time (why are they different in the first place?), but that should be much smaller overhead than hitting a threadlocal

Reply:
active interpreter TLS pointer - like this idea too, i considered doing this though that would expose some parts of the JIT interpreter, but it could also simplify the rest of the code